### ISSUE TITLE
Code 002

### ISSUE THREAD

**Issue by elvianod982** [NexiusTailer/Nex-AC#145]
A player sent me a video of what is apparently a problem they have experienced more than once:

https://www.youtube.com/watch?v=fEfKnZ3Om_A

As you can see, they were a passenger in a vehicle, which they left, and then got kicked by the anti-cheat:

22/03/2020 14:49:02 [Nex-AC debug] Veh model: 579, dist: 254.954330
22/03/2020 14:49:02 [Nex-AC] ID 11 appears suspicious. Reason code: 002 (5)
22/03/2020 14:49:02 OnPlayerDisconnect(11)
22/03/2020 14:49:02 [part] [679]Nas has left the server (11:2)

Is this a known issue or might it be a problem with my gamemode?

**Comment by NexiusTailer** [comment]
Judging by the logs, the distance between player’s new position onfoot and the current position of the vehicle he left from is more than 250 units (meters, anything else) which is quite strange.

**Comment by elvianod982** [comment]
There was another occurrence of 002 (5) today, some more info that might help perhaps:

**Log:**

> 23/03/2020 16:13:04 [Nex-AC debug] Veh model: 499, driver AFK time: 49, dist: 58.017765
> 23/03/2020 16:13:04 [Nex-AC] ID 15 appears suspicious. Reason code: 002 (5)
> 23/03/2020 16:13:04 OnPlayerDisconnect(15)
> 23/03/2020 16:13:04 [part] vDox has left the server (15:2)

**From the player in question:**
> I got kicked by the anti cheat, 002
> It happens when you are a passenger with a driver who lags and is warping around
> When I jumped out I got kicked

**Comment by NexiusTailer** [comment]
What ping and packetloss have the driver and the passenger?

**Comment by elvianod982** [comment]
I don't know the answer to that unfortunately.

However, some updates since this incident:

The player was connected over WiFi and I advised they use an Ethernet cable instead to improve their connection, which they are now doing.

Additionally I am now using Pawn.RakNet and from what I understand at https://forum.sa-mp.com/showthread.php?t=579963 this should also help:

"Added compatibility with Pawn.RakNet (for better desync before kick)"

I have also added a new client message when players are kicked to inform them to close all programs (including downloads) when playing to keep their internet as stable as possible, so I hope going forward players will be better educated, since I am sure this is not a Nex-AC problem but rather their internet which is unstable.
